src/components/PriceFilter.jsx

```diff
@@ -0,0 +1,42 @@
+import React, { useState } from "react";
+
+function PriceFilter({ updatePriceFilter, clearPriceFilter }) {
+    const [selectedFilters, setSelectedFilters] = useState([]);
+
+    const togglePriceFilter = (filterValue) => {
+        const filterIndex = selectedFilters.findIndex(f => f.min === filterValue.min && f.max === filterValue.max);
+        if (filterIndex !== -1) {
+            setSelectedFilters(prevFilters => prevFilters.filter((_, index) => index !== filterIndex));
+        } else {
+            setSelectedFilters(prevFilters => [...prevFilters, filterValue]);
+        }
+    };
+
+    const isFilterActive = (filterValue) => {
+        return selectedFilters.some(f => f.min === filterValue.min && f.max === filterValue.max);
+    };
+
+    const applyFilters = () => {
+        updatePriceFilter(selectedFilters);
+    };
+
+    const clearFilters = () => {
+        setSelectedFilters([]);
+        clearPriceFilter();
+    };
+
+    return (
+        <div>
+            <div className="flex justify-center space-x-4 mt-4">
+                <button className={`${isFilterActive({ min: 0, max: 50 }) ? "border rounded-md bg-primary-200 border-primary-200 text-white " : "border rounded-md bg-white border-primary-200 text-primary-200 " } hover:bg-primary-200 hover:text-white py-2 px-4 rounded`} onClick={() => togglePriceFilter({ min: 0, max: 50 })}> Under $50 </button>
```

Comment:
I would improve the price ranges otherwise the same gift appears in both price ranges.
Eg: If gift = $50 with the boundary you have, it appears in both category prices, under $50 and between 50-100.
And what if we have gifts with cents? So boundaries need to be better define.
Eg: I would say from 0.01-49.99 (Under $50); from 50-100.99 (second categ) from 101-250.99 (3rd catg) and the Over $250 which in the code will represented starting from 251-  so one gift will only fall into one categ

Ping me if I confused you!